FWCore/ParameterSet/test/parameterSetDescription_t.cc

```diff
@@ -924,172 +924,200 @@ namespace testParameterSetDescription {
-      // should have empty ParameterSets, but they have parameter "y"
-      try {
-        psetDesc15.validate(psetC);
-        assert(0);
-      } catch (edm::Exception const&) { /* There should be an exception */
+      {
+        edm::ParameterSetDescription psetDesc;
+        psetDesc.labelsFrom<std::vector<edm::ParameterSet>>("allowedLabelsC", edm::ParameterSetDescription());
+        // Now it should fail because the description says the contained vector<ParameterSet>
+        // should have empty ParameterSets, but the ParameterSets have parameter "y"
+        try {
+          psetDesc.validate(pset);
+          assert(0);
+        } catch (edm::Exception const&) { /* There should be an exception */
+        }
       }
 
       // Now include "y" in the description and it should once again pass validation
-      edm::ParameterSetDescription psetDesc6;
-      edm::ParameterSetDescription psetDesc7;
-      psetDesc7.add<int>("y", 1);
-
-      psetDesc6.labelsFrom<std::vector<edm::ParameterSet>>("allowedLabelsC", psetDesc7);
-      psetDesc6.validate(psetC);
-
-      // A minor variation, repeat with a string argument
-      edm::ParameterSetDescription psetDesc18;
-      edm::ParameterSetDescription psetDesc19;
-      edm::ParameterSetDescription psetDesc20;
-      psetDesc18.labelsFrom<std::vector<edm::ParameterSet>>(std::string("allowedLabelsC"));
-      psetDesc19.labelsFrom<std::vector<edm::ParameterSet>>(std::string("allowedLabelsC"),
+      edm::ParameterSetDescription nestedPSetDesc;
+      nestedPSetDesc.add<int>("y", 1);
+
+      {
+        edm::ParameterSetDescription psetDesc;
+        psetDesc.labelsFrom<std::vector<edm::ParameterSet>>("allowedLabelsC", nestedPSetDesc);
+        psetDesc.validate(pset);
+      }
+
+      // Minor variations, repeat with a string argument
+      {
+        edm::ParameterSetDescription psetDesc;
+        psetDesc.labelsFrom<std::vector<edm::ParameterSet>>(std::string("allowedLabelsC"));
+        psetDesc.validate(pset);
+      }
+      {
+        edm::ParameterSetDescription psetDesc;
+        psetDesc.labelsFrom<std::vector<edm::ParameterSet>>(std::string("allowedLabelsC"),
                                                             edm::ParameterSetDescription());
-      psetDesc20.labelsFrom<std::vector<edm::ParameterSet>>(std::string("allowedLabelsC"), psetDesc7);
-      psetDesc18.validate(psetC);
-      try {
-        psetDesc19.validate(psetC);
-        assert(0);
-      } catch (edm::Exception const&) { /* There should be an exception */
+        try {
+          psetDesc.validate(pset);
+          assert(0);
+        } catch (edm::Exception const&) { /* There should be an exception */
+        }
+      }
+      {
+        edm::ParameterSetDescription psetDesc;
+        psetDesc.labelsFrom<std::vector<edm::ParameterSet>>(std::string("allowedLabelsC"), nestedPSetDesc);
+        psetDesc.validate(pset);
       }
-      psetDesc20.validate(psetC);
```

Comment:
NOTE: At some future point, this really should be moved to either use an actual unit test framework.